components/datetime/src/mock/timezone.rs

```diff
@@ -0,0 +1,297 @@
+            time_variant,
+            country_code,
+        }
+    }
+
+    pub fn try_new(
+        gmt_offset: GmtOffset,
+        time_zone_id: Option<String>,
+        metazone_id: Option<String>,
+        time_variant: Option<String>,
+        country_code: Option<String>,
+    ) -> Result<Self, DateTimeError> {
+        Ok(Self {
+            gmt_offset,
+            time_zone_id,
+            metazone_id,
+            time_variant,
+            country_code,
+        })
+    }
+}
+impl FromStr for MockTimeZone {
+    type Err = DateTimeError;
+    fn from_str(input: &str) -> Result<Self, Self::Err> {
+        let gmt_offset = GmtOffset::from_str(input)?;
+        Ok(Self {
+            gmt_offset,
+            time_zone_id: None,
+            metazone_id: None,
+            time_variant: None,
+            country_code: None,
+        })
+    }
+}
+
+impl TimeZoneInput for MockTimeZone {
+    fn gmt_offset(&self) -> GmtOffset {
+        self.gmt_offset
+    }
+
+    fn time_zone_id(&self) -> Option<&str> {
+        self.time_zone_id.as_ref().map(AsRef::as_ref)
+    }
+
+    fn metazone_id(&self) -> Option<&str> {
+        self.metazone_id.as_ref().map(AsRef::as_ref)
+    }
+
+    fn time_variant(&self) -> Option<&str> {
+        self.time_variant.as_ref().map(AsRef::as_ref)
+    }
+
+    fn country_code(&self) -> Option<&str> {
+        self.country_code.as_ref().map(AsRef::as_ref)
+    }
+}
+
+/// Determines which ISO-8601 format should be used to format a `GmtOffset`.
+pub(crate) enum IsoFormat {
+    /// ISO-8601 Basic Format.
+    /// Formats zero-offset numerically.
+    /// e.g. +0500, +0000
+    Basic,
+
+    /// ISO-8601 Extended Format.
+    /// Formats zero-offset numerically.
+    /// e.g. +05:00, +00:00
+    Extended,
+
+    /// ISO-8601 Basic Format.
+    /// Formats zero-offset with the ISO-8601 UTC indicator: "Z"
+    /// e.g. +0500, Z
+    UtcBasic,
+
+    /// ISO-8601 Extended Format.
+    /// Formats zero-offset with the ISO-8601 UTC indicator: "Z"
+    /// e.g. +05:00, Z
+    UtcExtended,
+}
+
+/// Whether the minutes field should be optional or required in ISO-8601 format.
+pub(crate) enum IsoMinutes {
+    /// Minutes are always displayed.
+    Required,
+
+    /// Minutes are displayed only if they are non-zero.
+    Optional,
+}
+
+/// Whether the seconds field should be optional or excluded in ISO-8601 format.
+pub(crate) enum IsoSeconds {
+    /// Seconds are displayed only if they are non-zero.
+    Optional,
+
+    /// Seconds are not displayed.
+    None,
+}
+
+/// Whether a field should be zero-padded in ISO-8601 format.
+pub(crate) enum ZeroPadding {
+    /// Add zero-padding.
+    On,
+
+    /// Do not add zero-padding.
+    Off,
+}
+
+/// The GMT offset in seconds for a `ZonedDateTime`.
+#[derive(Copy, Clone, Debug, Default)]
+pub struct GmtOffset(i32);
+
+/// Get the ascii byte of a numeric digit.
+/// Only called on digits from in range 0..=9.
+const fn ascii_digit(digit: u8) -> u8 {
```

Comment:
It actually does not. This is a good catch. 

From the spec:
> "The digits used for hours, minutes and seconds fields in this format are the locale's default decimal digits"

We currently just use ASCII digits in `format_number()` for `DateTime`.

https://github.com/unicode-org/icu4x/blob/653c3ec23b76715caf9e50bc248828fac2049ee2/components/datetime/src/format.rs#L82-L100

Do we have code that gets localized digits? This might not be a trivial change if we need to pull in extra CLDR data for digit formatting. 

Do you consider this a blocker?

Reply:
#597 tracks fixing DateTimeFormat to use the local numbering system.

So no, it's not a blocker, but you should design your code to be ready for when that is implemented.  For example, TinyStr is kind-of not an option, b/c it doesn't support non-ASCII.